code/game/objects/items/devices/chameleonproj.dm

```diff
@@ -78,6 +88,9 @@
 	else
 		playsound(get_turf(src), 'sound/effects/pop.ogg', 100, 1, -6)
 		var/obj/effect/dummy/chameleon/C = new/obj/effect/dummy/chameleon(user.drop_location())
+		for(var/overlay_index in vis_overlay_data)
+			var/list/overlay_data = vis_overlay_data[overlay_index]
+			SSvis_overlays.add_vis_overlay(C, overlay_data["icon"], overlay_data["icon_state"], overlay_data["layer"], overlay_data["plane"], C.dir, alpha = overlay_data["alpha"], add_appearance_flags = overlay_data["appearance_flags"])
```

Comment:
Are these properly cleaned up on toggle off?

Reply:
The dummy that has these visual overlays applied on it is qdeleted when the projector is deactivated. I haven't dived too deep into the garbage collector, but from how visual overlays are implemented on things like holosigns and plastic flaps, deleting the object the overlays are on should clean them up.